File: packages/flutter_gemma_mediapipe/android/src/main/kotlin/dev/flutterberlin/flutter_gemma_mediapipe/FlutterGemmaMediaPipePlugin.kt
package dev.flutterberlin.flutter_gemma_mediapipe

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel

/** FlutterGemmaMediaPipePlugin — hosts the MediaPipe (.task) PlatformService
 *  HostApi + the "flutter_gemma_stream" async-result EventChannel. */
class FlutterGemmaMediaPipePlugin : FlutterPlugin {
  private var service: PlatformServiceImpl? = null
  private lateinit var eventChannel: EventChannel

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    val svc = PlatformServiceImpl(binding.applicationContext)
    service = svc
    eventChannel = EventChannel(binding.binaryMessenger, "flutter_gemma_stream")
    eventChannel.setStreamHandler(svc)
    PlatformService.setUp(binding.binaryMessenger, svc)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    PlatformService.setUp(binding.binaryMessenger, null)
    eventChannel.setStreamHandler(null)
    service?.cleanup()
    service = null
  }
}

File: packages/flutter_gemma_mediapipe/ios/Classes/FlutterGemmaMediaPipePlugin.swift
import Flutter

@available(iOS 13.0, *)
public class FlutterGemmaMediaPipePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let platformService = PlatformServiceImpl()
    PlatformServiceSetup.setUp(binaryMessenger: registrar.messenger(), api: platformService)

    let eventChannel = FlutterEventChannel(
      name: "flutter_gemma_stream", binaryMessenger: registrar.messenger())
    eventChannel.setStreamHandler(platformService)
  }
}
